src/main/java/core/basesyntax/impl/StorageImpl.java

```diff
@@ -1,19 +1,73 @@
 package core.basesyntax.impl;
 
 import core.basesyntax.Storage;
+import java.util.Arrays;
 
 public class StorageImpl<K, V> implements Storage<K, V> {
+    private static final int MAX_SIZE = 10;
+    private Object[] keys;
+    private Object[] values;
+    private boolean hasNullKey;
+    private V nullKeyValue;
+    private int size;
+
+    @SuppressWarnings("unchecked")
+    public StorageImpl() {
+        this.keys = new Object[MAX_SIZE];
+        this.values = new Object[MAX_SIZE];
+        this.hasNullKey = false;
+        this.nullKeyValue = null;
+        this.size = 0;
+    }
+
     @Override
     public void put(K key, V value) {
+        if (key == null) {
```

Comment:
```suggestion
        int index = indexOf(key);
        if (index != NO_INDEX) {
            values[index] = value;
        } else if (index == -1) {
            keys[size] = key;
            values[size] = value;
            size++;
        }
```

implement indexOf method